integration/setup/setup.go

```diff
@@ -0,0 +1,79 @@
+// +build k8srequired
+
+package setup
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+
+	"github.com/giantswarm/app-operator/integration/env"
+	"github.com/giantswarm/app-operator/integration/teardown"
+)
+
+func WrapTestMain(ctx context.Context, h *framework.Host, crdClient *k8scrdclient.CRDClient, helmClient *helmclient.Client, l micrologger.Logger, m *testing.M) {
+	var v int
+	var err error
+
+	err = h.CreateNamespace("giantswarm")
+	if err != nil {
+		log.Printf("%#v\n", err)
+		v = 1
+	}
+
+	err = helmClient.EnsureTillerInstalled(ctx)
+	if err != nil {
+		log.Printf("%#v\n", err)
+		v = 1
+	}
+
+	err = resources(ctx, h, crdClient, l)
+	if err != nil {
+		log.Printf("%#v\n", err)
+		v = 1
+	}
+
+	if v == 0 {
+		v = m.Run()
+	}
+
+	if env.KeepResources() != "true" {
+		// only do full teardown when not on CI
+		if env.CircleCI() != "true" {
+			err := teardown.Teardown(h, helmClient)
+			if err != nil {
+				log.Printf("%#v\n", err)
+				v = 1
+			}
+			// TODO there should be error handling for the framework teardown.
+			h.Teardown()
+		}
+	}
+
+	os.Exit(v)
+}
+
+func resources(ctx context.Context, h *framework.Host, crdClient *k8scrdclient.CRDClient, l micrologger.Logger) error {
+	var err error
+
+	err = crdClient.EnsureCreated(ctx, v1alpha1.NewChartCRD(), backoff.NewExponential(backoff.ShortMaxWait, backoff.ShortMaxInterval))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = h.InstallStableOperator("app-operator", "app", "")
```

Comment:
`InstallStableOperator` is the wrong method here as it will install the `master` branch of your operator.
Where did you take this boilerplate from?

The goto way to do this should be like in `aws-operator`: https://github.com/giantswarm/aws-operator/blob/master/integration/setup/setup.go#L118 
This uses the release package which you can find here:
https://github.com/giantswarm/e2e-harness/blob/master/pkg/release/release.go#L286

The `release` package uses `helmClient` inside to make creation more graceful and give you better error handling.

Reply:
See especially the `release.NewVersion(env.CircleSHA())` to obtain the correct version for your branch operator.